java-squid/src/main/java/org/sonar/java/resolve/Types.java

```diff
@@ -66,15 +66,14 @@ public boolean isSubtype(JavaType t, JavaType s) {
           result = isSubtype(((JavaType.ArrayJavaType) t).elementType(), ((JavaType.ArrayJavaType) s).elementType());
           break;
         case JavaType.CLASS:
-          if(t.tag != s.tag) {
-            result = false;
-            break;
-          }
           result = t.isSubtypeOf(s);
           break;
         case JavaType.BOT:
           result = s.tag == JavaType.BOT || s.tag == JavaType.CLASS || s.tag == JavaType.ARRAY;
           break;
+        case JavaType.WILDCARD:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This case's code block is the same as the block for the case on line 68. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1871)